pyartcd/pyartcd/pipelines/build_rhcos.py

```diff
@@ -58,184 +52,72 @@ def __init__(self, runtime: Runtime, new_build: bool, ignore_running: bool, vers
-                result=b["result"],
-            ) for b in self.request_session.get(
-                f"{JENKINS_BASE_URL}/job/{job}/api/json?tree=builds[number,description,result]"
-            ).json()["builds"]
-            if b["number"] == number
-        ), None)
-
-    def wait_for_builds(self):
-        """Wait for all builds for this version to complete, and give status updates on stderr"""
-        builds_seen: Dict[Tuple, str] = {}
-        completed_builds: Dict[Tuple, Dict] = {}
-        for _ in range(2880):  # x 10s = about 8 hours (slower if bad/no response)
-            new_builds_seen: Dict[Tuple, str] = {
-                (b["job"], b["number"]): b["description"] or "[no description yet]"
-                for b in self.query_existing_builds()
-            }
-
-            # check if any previous builds have newly completed
-            for spec in builds_seen.keys():
-                if spec not in new_builds_seen and spec not in completed_builds:
-                    completed_builds[spec] = completed = self.build_result(*spec)
-                    if completed:  # silently ignore if it's somehow not there... should never happen
-                        print(f"{completed['url']} finished with {completed['result']}: '{completed['description']}'", file=sys.stderr)
-
-            # if there are no builds left running, we're done
-            if not new_builds_seen:
-                break
-
-            # check if there are new or changed builds
-            for spec, description in new_builds_seen.items():
-                job, number = spec
-                if spec not in builds_seen:
-                    print(f"New build #{number} of {job} job: {self.build_url(job, number)}", file=sys.stderr)
-                elif description != builds_seen[spec]:
-                    print(f"Build #{number} of {job} job update: '{description}'", file=sys.stderr)
-
-            builds_seen = new_builds_seen
-            time.sleep(10)
-        else:  # only gets here if the for loop reaches the count
-            raise Exception("Waited too long for builds to complete")
-
-        return list(completed_builds.values())
+        job_url = f"{JENKINS_BASE_URL}/job/{self.job}/buildWithParameters"
+        params = dict(STREAM=self._stream, EARLY_ARCH_JOBS="false", FORCE=self.new_build)  # this is build job parameter
+        build_number = self.trigger_build(job_url, params)
+        build = self.request_session.get(f"{JENKINS_BASE_URL}/job/{self.job}/{build_number}/api/json").json()
+        return dict(url=build['url'], result=build['result'], description=build['description'])
+
+    def trigger_build(self, job_url, params):
+        """trigger build and return build bumber"""
+        response = self.request_session.post(job_url, data=params)
+        if response.status_code not in (201, 200):
+            raise Exception(f"Failed to trigger rhcos build: {response}")
+        queue_url = response.headers.get('Location')
```

Comment:
now we know trigger a jenkins job will return a queue, and wait until the queue start it will get a build id